src/svd/bindings.py

```diff
@@ -0,0 +1,988 @@
+    )
+
+    def __iter__(self) -> Iterator[SauRegion]:
+        """Iterate over all predefined SAU regions."""
+        return iter_element_children(self, SauRegion.TAG)
+
+    # (internal) SAU regions.
+    _region: SauRegion = elem("region", SauRegion)
+
+
+@binding
+class Cpu(objectify.ObjectifiedElement):
+    """Description of the device processor"""
+
+    TAG = "cpu"
+
+    # CPU name. See CpuName for possible values.
+    name: CpuName = elem("name", CpuNameElement)
+
+    # CPU hardware revision with the format "rNpM".
+    revision: str = elem("revision", StringElement)
+
+    # Default endianness of the CPU.
+    endian: Endian = elem("endian", EndianElement)
+
+    # True if the CPU has a memory protection unit (MPU).
+    has_mpu: Optional[bool] = elem("mpuPresent", BoolElement, default=None)
+
+    # True if the CPU has a floating point unit (FPU).
+    has_fpu: Optional[bool] = elem("fpuPresent", BoolElement, default=None)
+
+    # True if the CPU has a double precision floating point unit (FPU).
+    fpu_is_double_precision: Optional[bool] = elem("fpuDP", BoolElement, default=None)
+
+    # True if the CPU implements the SIMD DSP extensions.
+    has_dsp: Optional[bool] = elem("dspPresent", BoolElement, default=None)
+
+    # True if the CPU has an instruction cache.
+    has_icache: Optional[bool] = elem("icachePresent", BoolElement, default=None)
+
+    # True if the CPU has a data cache.
+    has_dcache: Optional[bool] = elem("dcachePresent", BoolElement, default=None)
+
+    # True if the CPU has an instruction tightly coupled memory (ITCM).
+    has_ictm: Optional[bool] = elem("itcmPresent", BoolElement, default=None)
+
+    # True if the CPU has a data tightly coupled memory (DTCM).
+    has_dctm: Optional[bool] = elem("dtcmPresent", BoolElement, default=None)
+
+    # True if the CPU has a Vector Table Offset Register (VTOR).
+    has_vtor: bool = elem("vtorPresent", BoolElement, default=True)
+
+    # Bit width of interrupt priority levels in the Nested Vectored Interrupt Controller (NVIC).
+    num_nvic_priority_bits: int = elem("nvicPrioBits", SvdIntElement)
+
+    # True if the CPU has a vendor-specific SysTick Timer.
+    # If False, the Arm-defined System Tick Timer is used.
+    has_vendor_systick: bool = elem("vendorSystickConfig", BoolElement)
+
+    # Maximum interrupt number in the CPU plus one.
+    num_interrupts: Optional[int] = elem(
+        "deviceNumInterrupts", SvdIntElement, default=None
+    )
+
+    # Number of supported Secure Attribution Unit (SAU) regions.
+    num_sau_regions: int = elem("sauNumRegions", SvdIntElement, default=0)
+
+    # Predefined Secure Attribution Unit (SAU) regions, if any.
+    preset_sau_regions: Optional[SauRegionsConfig] = elem(
+        "sauRegionsConfig", SauRegionsConfig, default=None
+    )
+
+
+@binding
+class AddressBlock(objectify.ObjectifiedElement):
+    """Address range mapped to a peripheral."""
+
+    TAG = "addressBlock"
```

Comment:
nit: `TAG` attributes further down are type-hinted, but missing here and some other classes like `FieldElement`.